Add official docker image

gfs opened the pull request:
Fix #64.

This adds a Dockerfile for DevSkim and and the pipeline steps to build it. Missing are the set-up connections for a docker repository for pushing the image somewhere.  For now we just build it.

Harvester57 commented:
This Dockerfile doesn't work as-is, at least with BinSkim version 0.4.105. I had to add the libicu60 package to the image, otherwise the devskim binary would crash with a "Process terminated. Couldn't find a valide ICU package installed on the system."

I added the following lines after the COPY directive:

`RUN apt update`
`RUN apt install libicu60 -y`

Works like a charm now :)

gfs commented:
Thanks! I thought I had fixed this localization issue in the csproj but this should fix it.

Sent from ProtonMail mobile

-------- Original Message --------
On Feb 27, 2020, 5:20 AM, Harvester wrote:

> This Dockerfile doesn't work as-is, at least with BinSkim version 0.4.105. I had to add the libicu60 package to the image, otherwise the devskim binary would crash with a "Process terminated. Couldn't find a valide ICU package installed on the system."
>
> I added the following lines after the COPY directive:
>
> RUN apt update
> RUN apt install libicu60 -y
>
> Works like a charm now :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, [view it on GitHub](https://github.com/microsoft/DevSkim/pull/94?email_source=notifications&email_token=AAAYEVHJNL5RJFIZ2F5Q7YTRE64Y3A5CNFSM4KUE2JB2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENELD4Q#issuecomment-591966706), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AAAYEVHIPDLX5AYKMD2UTGLRE64Y3ANCNFSM4KUE2JBQ).
